utils/large_buffer_test.go

```diff
@@ -0,0 +1,162 @@
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLargeBufferHappy(t *testing.T) {
+	for testName, tc := range map[string]struct {
+		chunks                [][]byte
+		expectedBufferedBytes []byte
+		expectedFileBytes     []byte
+		expectedReadBackBytes []byte
+	}{
+		"happy memory buffered": {
+			chunks:                [][]byte{[]byte("123")},
+			expectedBufferedBytes: []byte("123"),
+			expectedReadBackBytes: []byte("123"),
+		},
+		"happy memory plus file": {
+			chunks: [][]byte{
+				[]byte("0123456789ABCDEF"),
+				[]byte("XYZ"),
+			},
+			expectedBufferedBytes: []byte("0123456789ABCDEF"),
+			expectedFileBytes:     []byte("=BSf"),
+			expectedReadBackBytes: []byte("0123456789ABCDEFXYZ"),
+		},
+	} {
+		testfunc := func(t *testing.T, lbuf *LargeBuffer) {
+			a := assert.New(t)
+			for _, chunk := range tc.chunks {
+				n, err := lbuf.Write(chunk)
+				a.Empty(err, "Expected no error")
+				a.Equal(len(chunk), n, "Wrong n returned (bytes written)")
+			}
+
+			a.Equal(lbuf.buffer.Bytes(), tc.expectedBufferedBytes, "Buffered contents differs")
+
+			err := lbuf.Close()
+			a.Empty(err, "Expected no error")
```

Comment:
Nit: Why not use `a.Nil()` and be explicit what we expect?

Reply:
Or better yet, `a.NoError()`?